File: Windows Native Runtime/SceneManager.cpp
#include "SceneManager.h"

static std::list<Scene> scene_array;

void SceneManager::addScene(Scene scene)
{
	scene_array.push_back(scene);
}

Scene SceneManager::getScene(std::string scene_name)
{
	for (std::list<Scene>::iterator cur_scene = scene_array.begin(); cur_scene != scene_array.end(); cur_scene++)
	{
		if (cur_scene->name == scene_name)
		{
			return *cur_scene;
		}
	}
}

File: Java Desktop Runtime/jruntime/SceneManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package jruntime;

import java.util.*;

/**
 *
 * @author Riju
 */
public class SceneManager 
{
     static Vector scene_array = new Vector( );
		 
	 public static void addScene(Scene scene )
		 {
                       scene_array.addElement(scene);
		 }
		 
		 public static Scene getScene(String name)
		 {
		     for(int cntr = 0;cntr < scene_array.size( );cntr++)
			 {
				if (((Scene) scene_array.elementAt(cntr)).name == name)
				{
					return (Scene) scene_array.elementAt(cntr);
				}
                                
			 }
		
			 return null;
		 }
                 
}
